### ISSUE TITLE
Custom Empty State View

### ISSUE THREAD

**Issue by silviuandrica** [luispadron/UIEmptyState#12]
Hi @luispadron ,

Im trying to use `UIEmptyState` with a custom empty view for a `UITableViewController`, but I cant manage to make it show.

The delegate function `emptyStateViewDidShow` is called, but the view isnt there. Any ideas?

Thanks,
Silviu

**Comment by luispadron** [comment]
Are you returning anything for the data source property `emptyStateView`? By default this returns a UIEmptyStateView, but you can return any subclass of UIView and it should work.

**Comment by silviuandrica** [comment]
Hey,

Thanks for answering me so quickly.

My `viewDidLoad` method contains the following

```
let emptyView : EmptyStateView = EmptyStateView.fromNib()
self.emptyStateView = emptyView
self.emptyStateDataSource = self
 self.emptyStateDelegate = self
```

And 
```
func emptyStateViewDidShow(view: UIView){
        print("Did show \(view)")
}
```

prints out `Did show <....EmptyStateView: 0x106135cd0; frame = (0 0; 375 667); autoresize = W+H; layer = <CALayer: 0x1d0239c00>>`...

Inspecting in the Xcode debugger, the Quick view shows me the right view... it just doesnt show up in my uitableviewcontroller.
 
Cheers,
Silviu

**Comment by silviuandrica** [comment]
In the `emptyStateViewDidShow` method, should `view.superview` be nil?

**Comment by luispadron** [comment]
You don’t want to set the emptyStateView property outright. 

The data source computed variable `emptyStateView` is the one that gets called. 

I recommend reading the docs so you can better understand how it works, look at specifically the UIEmptyStateDatasource [here](http://htmlpreview.github.io/?https://github.com/luispadron/UIEmptyState/blob/master/docs/index.html)

**Comment by silviuandrica** [comment]
Thanks. I am a bit new to Swift, so I may be doing weird things...

But, I tried 

```
public var emptyStateView: UIView {
        get {
            let emptyView : EmptyStateView = EmptyStateView.fromNib()
            return emptyView
        }
    }
```

Which is similar to the `UIEmptyStateDataSource` extension, but I have a compile issue: `Overriding non-open var outside of its defining module`. Any help is greatly appreciated...

**Comment by silviuandrica** [comment]
Hey,

I solved my problem by creating a UIEmptyStateDataSource extension.

Thanks for all your help. 

May I suggest you specify this in the documentation?

Cheers,
Silviu

**Comment by luispadron** [comment]
Well, looks like I messed up by naming two properties the same. 
Thanks for bringing this to my attention, I'm going to push a new version out soon and it should work as expected.

**Comment by silviuandrica** [comment]
Glad to be of help :-)

**Comment by luispadron** [comment]
Fixed in [version 3.0.0](https://github.com/luispadron/UIEmptyState/releases/tag/v3.0.0)

You can create a custom view like this in your `UIViewController` class.

```swift
var emptyStateView: UIView {
    let view = UIView()
    // Modify the view, etc
    view.backgroundColor = .blue
    // Important to allow constraints since by default the library will add center x and y.
    view.translatesAutoResizingMaskIntoConstraints = false
    // Add any extra constraints such as height and width
    view.heightAnchor.constraint(withConstant: 100).isActive = true
    view.widthAnchor.constraint(withConstant: 100).isActive = true
    return view
}
```
